config/routes.rb

```diff
@@ -1,3 +1,16 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  
+  namespace :api do
+    namespace :v1 do
+      get '/invoices/find', to: 'invoices/find#show'
```

Comment:
nice! I was wondering how the routing would look for the single find compared to the find_all, and this makes sense.